File: 1411-convert-binary-number-in-a-linked-list-to-integer/convert-binary-number-in-a-linked-list-to-integer.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    int getDecimalValue(ListNode* head) {
        ListNode* temp =head;
        int count=-1;
        int ans=0;
        while(temp!=NULL) {
            temp=temp->next;
            count++;
        }
        ListNode* pos=head;
        while(pos!=NULL) {
            int res = pos->val;
            pos=pos->next;
            ans += res * pow(2, count);
            count--;
        }
        return ans;  
    }
};

File: 1411-convert-binary-number-in-a-linked-list-to-integer/convert-binary-number-in-a-linked-list-to-integer.dart
/**
 * Definition for singly-linked list.
 * class ListNode {
 *   int val;
 *   ListNode? next;
 *   ListNode([this.val = 0, this.next]);
 * }
 */

class Solution {
  int getDecimalValue(ListNode? head) {
    int ans = 0;
    ListNode? temp = head;

    while (temp != null) {
      ans = ans * 2 + temp.val;
      temp = temp.next;
    }

    return ans;
  }
}